packages/api/tests/unit/test_signin_failure_activity.py

```diff
@@ -0,0 +1,76 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+import pytest
+from microsoft.teams.api.activities.invoke.sign_in import SignInFailureInvokeActivity
+from microsoft.teams.api.models import Account, ConversationAccount, SignInFailure
+
+
+@pytest.mark.unit
+class TestSignInFailureInvokeActivity:
+    """Unit tests for SignInFailureInvokeActivity class."""
+
+    def test_should_create_signin_failure_activity(self) -> None:
+        """Test creating a signin failure activity with code and message."""
+        failure = SignInFailure(code="resourcematchfailed", message="Resource match failed")
+        user = Account(id="user-1", name="Test User")
+        bot = Account(id="bot-1", name="Test Bot")
+        conversation = ConversationAccount(id="conv-1")
+
+        activity = SignInFailureInvokeActivity(
+            id="activity-1",
+            name="signin/failure",
+            value=failure,
+            from_=user,
+            conversation=conversation,
+            recipient=bot,
+        )
+
+        assert activity.name == "signin/failure"
+        assert activity.type == "invoke"
+        assert activity.value.code == "resourcematchfailed"
+        assert activity.value.message == "Resource match failed"
+
+    def test_should_deserialize_signin_failure_activity(self) -> None:
```

Comment:
Can you move this test to a new test file called "text_fixtures". The point of that suite will be to test the deserialization of various fixtures. You can put the fixtures in tests/fixtures folder. Set it up so that the user just needs to point to the fixture file, and it should correctly deserialize. It's okay that the first test will be just this activity type.